bokeh/util/api_crawler.py

```diff
@@ -0,0 +1,232 @@
+        files_diff = self._operation(set(self.former), set(self.latter))
+
+        # Diff files
+        for x in intersection.keys():
+            if x in files_diff:
+                diff[x] = {}
+        return intersection, diff
+
+    def diff_functions_classes(self, diff, intersection):
+        for x in intersection.keys():
+            former_items = self.former.get(x)
+            latter_items = self.latter.get(x)
+            if former_items and latter_items:
+                function_diff = self._operation(
+                    set(former_items["functions"]),
+                    set(latter_items["functions"])
+                )
+                class_diff = self._operation(
+                    set(list(former_items["classes"].keys())),
+                    set(list(latter_items["classes"].keys()))
+                )
+                if function_diff or list(class_diff):
+                    diff[x]= copy.deepcopy(intersection[x])
+                    if list(class_diff):
+                        diff_dict = {y: {} for y in class_diff}
+                        diff[x]["classes"] = diff_dict
+                    else:
+                        diff[x]["classes"] = {}
+
+                    if function_diff:
+                        diff[x]["functions"] = list(function_diff)
+                    else:
+                        diff[x]["functions"] = []
+        return diff
+
+    def diff_methods(self, diff, intersection):
+        for x in intersection.keys():
+            former_classes = self.former[x].get("classes") if self.former.get(x) else {}
+            latter_classes = self.latter[x].get("classes") if self.latter.get(x) else {}
+            if former_classes and latter_classes:
+                for y in intersection[x]["classes"]:
+                    # Prevent NoneType errors by returning empty dict.
+                    former_methods = self.former[x]["classes"].get(y, {})
+                    latter_methods = self.latter[x]["classes"].get(y, {})
+                    if former_methods.get("methods") and latter_methods.get("methods"):
+                        former_values = set(list(former_methods.values())[0])
+                        latter_values = set(list(latter_methods.values())[0])
+                        methods_diff = self._operation(former_values, latter_values)
+                        if methods_diff:
+                            if not diff.get(x):
+                                diff[x] = {}
+                                diff[x]["classes"] = {}
+                            diff[x]["classes"][y] = copy.deepcopy(intersection[x]["classes"][y])
+                            diff[x]["classes"][y]["methods"] = methods_diff
+        return diff
+
+    def diff_modules(self):
+        intersection, diff = self.diff_files()
+        diff = self.diff_functions_classes(diff, intersection)
+        diff = self.diff_methods(diff, intersection)
+        return diff
+
+    def parse_diff(self, diff):
```

Comment:
This name is confusing. I think this should be called `pretty_diff`, or something along this.